File: RelacjeSportowe.Web/ClientApp/src/app/models/dtos/requests/transmission-event-request.ts
abstract class TransmissionEventRequest {
  public transmissionId: number;
  public transmissionEventTypeId: number;
  public description: string;
  public timeInfo: string;

  constructor(data: any) {
    this.transmissionId = data.transmissionId;
    this.transmissionEventTypeId = data.transmissionEventTypeId;
    this.description = data.description;
    this.timeInfo = data.timeInfo;
  }
}

export class AddTransmissionEventRequest extends TransmissionEventRequest {
  constructor(data: any) {
    super(data);
  }
}

export class UpdateTransmissionEventRequest extends TransmissionEventRequest {
  public id: number;

  constructor(data: any) {
    super(data);
    this.id = data.id;
  }
}

File: RelacjeSportowe.DataAccess/Dtos/Requests/TransmissionEventRequest.cs
namespace RelacjeSportowe.DataAccess.Dtos.Requests
{
    public abstract class TransmissionEventRequest
    {
        public int TransmissionId { get; set; }

        public int TransmissionEventTypeId { get; set; }

        public string Description { get; set; }

        public string TimeInfo { get; set; }
    }

    public class AddTransmissionEventRequest : TransmissionEventRequest
    {
    }

    public class UpdateTransmissionEventRequest : TransmissionEventRequest
    {
        public int Id { get; set; }
    }
}
